modules/store-devtools/src/extension.ts

```diff
@@ -163,7 +169,31 @@ export class DevtoolsExtension {
     // Listen for lifted actions
     const liftedActions$ = changes$.pipe(
       filter(change => change.type === ExtensionActionTypes.DISPATCH),
-      map(change => this.unwrapAction(change.payload))
+      map(change => this.unwrapAction(change.payload)),
+      concatMap((action: any) => {
+        if (action.type === IMPORT_STATE) {
+          if (this.router) {
```

Comment:
I don't think the router should be integrated here. The route may need the state we are importing, but this prevents that from happening until the navigation is over.

Reply:
Maybe we can listen for any `@ngrx/store/update-reducers` actions and delay the import until none have been received within a span of time.

Reply:
`liftedActions$` retrieves the IMPORT_STATE-action before it is passed to the reducer. So as long as IMPORT_STATE is not emitted by `liftedActions$`, the import will not be triggered. `update-reducers` etc will be fired before the route is loaded because that happens on a different stream of actions (`actions$`), so there is no block. So in my opinion this is fine.

Reply:
I think it's not quite clear what I mean, so I'll expand on it:

### Situation 1: Router is used by user. User clicks the "import"-button of the redux devtools and manually imports a state
The added logic will not matter because  router.navigated will be true (the first navigation already happened) , therefore nothing to wait on, therefore the IMPORT action is fired straight away.

### Situation 2: Router is used by user. User turned on the "persist"-feature and reloads.
The added logic now matters. router.navigated will be false because the router is in the process of navigating when the action comes in. What that means is that the modules connected to that route are still getting instantiated, so for example components/reducers/effects for that route are not there yet.  So if we just let the IMPORT action pass through, the reducer would import the state too soon, some actions would get ignored because the reducers handling it do not exist yet. That's why we wait until the NavigationEnd-event of the router is fired. At this point we know all the initialization logic is done.<br>
There will be no blocking of state needed for the router because initial reducers/effects/state will be loaded during navigation. These things do not come through the `liftedActions$` stream, so blocking further events there until we are finished importing is not an issue. Also something like "oh you are not logged in, I will redirect you" is fine because, well, we just import the state after that and the application will get back to the correct route.

### Situation 3: Router is not used by user.
The whole router logic does not matter now, we wait 1 second just to make sure.

Reply:
I understand what's going on here, but I'd still rather support this in a consistent way without the router if possible.

Idea: After the IMPORT_STATE action comes in, wait some amount of time, then switch to listening for update-actions and debounce the IMPORT_STATE action. We can inject and listen to the store actions being dispatched.

Reply:
Like this?

```
concatMap((action: any) => {
  if(action.type === IMPORT_STATE) {
    return actions$.pipe(
      ofType(UPDATE), // '@ngrx/store/update-reducers'
      debounceTime(1000),
      map(() => action)
    );
  }
...
```

Reply:
We cannot know whether the IMPORT_STATE action is fired because of the persist option turned on or because of a manual import. But that's no harm, the user just has a minimal delay (of 1 second) until the manual import starts.

Reply:
Good point, a race condition / timeout in the likes of "there is no UPDATE action at all" that fires after 1 second is needed.
I will implement it and update the PR.